### ISSUE TITLE
Cache and vendor dir

### ISSUE THREAD

**Issue by balping** [nWidart/laravel-modules#429]
Hello,

I don't really understand the purpose of this code segment:

https://github.com/nWidart/laravel-modules/blob/8e86c2b0f0fa37b4684883c996d01c15bd9862e8/src/Laravel/Repository.php#L37-L48

If cache is enabled, it loops through all modules and replaces the path of each module. Why? The new path might or might not be the real one.

For instance I have a module in the vendor folder that is loaded correctly if caching is turned off.

Running `module:list` correctly finds it:

```
+------+---------+-------+------------------------------------------+
| Name | Status  | Order | Path                                     |
+------+---------+-------+------------------------------------------+
| Auth | Enabled | 0     | /project-root/vendor/balping/auth-module |
+------+---------+-------+------------------------------------------+
```

But when I enable caching, the above code tries to cache the path as `project-root/Modules/Auth` which is obviously wrong!

This leads to this error later:

```
[Illuminate\Contracts\Filesystem\FileNotFoundException]  
  File does not exist at path /module.json  
```

So my question is: what's the purpose of overriding the path?  @nWidart said several times in different issues that people use modules from the vendor dir. But do they do that with caching enabled? I haven't found any useful example on that.

**Comment by balping** [comment]
I checked and that piece of code was added in this commit: https://github.com/pingpong-labs/modules/commit/d084cf96f07f90e6a29724c77343392250bd3ecc to [fix a bug](https://github.com/pingpong-labs/modules/issues/85).

However there's no explanation for why it's necessary to replace the path – and a lot things have changed since, so that might  be not necessary. There's no test hitting that function, so I can't be sure that fixing this bug by replacing  [this line](https://github.com/nWidart/laravel-modules/blob/8e86c2b0f0fa37b4684883c996d01c15bd9862e8/src/Laravel/Repository.php#L42) wouldn't break anything.

**Comment by nWidart** [comment]
Hm interesting. Indeed that part is hard to test. I'm not using this feature specifically so I don't have an immediate explanation for this solution.

**Comment by balping** [comment]
I found the reason: only the content of `module.json` was cached. That doesn't contain the path of the module, so when retrieving from the cache, it was necessary to recalculate the path. But it was based on the assumption that the module is under the default folder in the project root.

Now I fixed the issue by caching the path too. I manually tested it and it works now both with modules in the vendor dir and modules in the default folder.

PS.: `Lumen/Repository.php` and `Laravel/Repository.php` are exact copies. It's unnecessary code duplication – now I had to change the same line twice. Is there a reason behind this duplication? Maybe the code could be moved back to the abstract parent repository.
